semver.go

```diff
@@ -374,18 +379,26 @@ func ParseTolerant(s string) (Version, error) {
 	}
 
 	// Generate version to properly parse
-	s = fmt.Sprintf("%s.%s.%s", majorStr, minorStr, patchStr)
+	b := strings.Builder{}
+	b.WriteString(majorStr)
+	b.WriteString(".")
+	b.WriteString(minorStr)
+	b.WriteString(".")
+	b.WriteString(patchStr)
 	if len(revisionStr) > 0 {
-		s = fmt.Sprintf("%s.%s", s, revisionStr)
+		b.WriteString(".")
+		b.WriteString(revisionStr)
```

Comment:
```suggestion
		fmt.Fprintf(&b, "%s.%s", s, revisionStr)
```

Reply:
This is adding the full version string back in to the string builder; pretty sure we'll need to keep the `WriteString` calls here to build the rest of the string.

Reply:
oh, right, it should be `fmt.Fprintf(&b, ".%s", revisionStr)`